tool/templates/js/enum.js.jinja

```diff
@@ -16,10 +13,8 @@ export class {{type_name}} {
     static getAllEntries() {
         return {{type_name}}.#values.entries();
     }
-
-    {# TODO: I think it's possible to allow for constructors,
-    but we just need to check if a constructor will be generated. -#}
-    constructor(value) {
+    
+    {% if overrides_constructor -%} #internalConstructor {%- else -%} constructor {%- endif -%}(value) {
```

Comment:
issue: I would unconditionally call this `internalConstructor`. constructors work differently from methods, we should not use the same code for both

All branches of this method should `return this` OR return an object.